platform_buckets_api/kube_storage.py

```diff
@@ -16,135 +27,267 @@
     StorageError,
 )
 from platform_buckets_api.utils.asyncio import asyncgeneratorcontextmanager
+from .kube_utils import (
+    ID_LABEL,
+    BUCKET_NAME_LABEL,
+    APOLO_ORG_NAME_LABEL,
+    APOLO_PROJECT_NAME_LABEL,
+    OWNER_LABEL,
+    CREDENTIALS_NAME_LABEL,
+)
 
 
 class K8SBucketsStorage(BucketsStorage):
-    def __init__(self, kube_api: KubeApi) -> None:
-        self._kube = kube_api
+    def __init__(self, kube_client: KubeClient) -> None:
+        self._kube_client = kube_client
+
+    def _bucket_to_k8s_model(self, bucket: BucketType) -> V1UserBucketCRD:
+        bucket_crd_mapper = BucketCRDMapper.to_primitive(bucket)
```

Comment:
I guess instead of `to_primitive() -> dict` we should use `to_model() -> V1UserBucketCRD` instead.

It makes the source a little better readable and eliminates *obj -> dict -> model* conversion, replacing it with just *obj -> model* one.